File: html/old/R2/SACO R2 2008 (Solutions)/tiles.cpp
// SACO 2nd Round 2008
// Tiles Solution by Keegan Carruthers-Smith

#include <iostream>

using namespace std;

int main() {
    int N;

    cout << "Enter N: ";
    cin >> N;

    int *nTiles = new int[N+1];
    nTiles[0] = 0;
    nTiles[1] = 0;
    nTiles[2] = 1;
    nTiles[3] = 1;
    
    for (int i = 4; i <= N; i++)
	nTiles[i] = (nTiles[i-2] + nTiles[i-3]) % 1007;

    cout << nTiles[N] << endl;

    delete[] nTiles;

    return 0;
}

File: html/old/backup/sacor2_2008/tiles.pas
{
	SACO Round 2 2008
	Solution to problem 6 - Tiles
	by Ben Steenhuisen
}
PROGRAM tiles;

var	tileto:array[1..2000000] of integer;
	x,N:integer;
BEGIN
	writeln("Enter N: ");read(N);
	tileto[1]:=0;tileto[2]:=1;tileto[3]:=1;
	for x:=4 to N do tileto[x]:=(tileto[x-3]+tileto[x-2]) mod 1007;
	writeln(tileto[N]);
END.
